include/dav1d/picture.rs

```diff
@@ -90,24 +91,43 @@ pub struct Dav1dPicture {
     pub mastering_display_ref: Option<RawArc<Rav1dMasteringDisplay>>, // opaque, so we can change this
     pub itut_t35_ref: Option<RawArc<DRav1d<Rav1dITUTT35, Dav1dITUTT35>>>, // opaque, so we can change this
     pub reserved_ref: [uintptr_t; 4],
-    pub r#ref: *mut Dav1dRef,
-    pub allocator_data: *mut c_void,
+    pub r#ref: Option<NonNull<Dav1dRef>>,
+    pub allocator_data: Option<NonNull<c_void>>,
 }
 
 #[derive(Clone)]
+pub(crate) struct Rav1dPictureData {
+    pub data: [*mut c_void; 3],
```

Comment:
It'll be a problem going forward since with an array, the compiler can see that `&mut`s of different compile-time indices are disjoint, but it won't be able to see that through an `impl Index`.  We also don't want unnecessary bounds checking of constant indices if something doesn't get inlined.